modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C

```diff
@@ -0,0 +1,150 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowAddMaterialJoiner.h"
+#include "AddMaterialAction.h"
+#include "AddUserObjectAction.h"
+
+registerMooseAction("PorousFlowApp", PorousFlowAddMaterialJoiner, "add_joiners");
+
+template <>
+InputParameters
+validParams<PorousFlowAddMaterialJoiner>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription(
+      "Adds PorousFlowJoiner materials as required for each phase-dependent property");
+  return params;
+}
+
+PorousFlowAddMaterialJoiner::PorousFlowAddMaterialJoiner(const InputParameters & params)
+  : Action(params)
+{
+}
+
+void
+PorousFlowAddMaterialJoiner::act()
+{
+  // This task only runs after the UserObject and material actions have been added,
+  // so we can get the name of the PorousFlowDictator UserObject and all material
+  // types that are added in the input file
+  if (_current_task == "add_joiners")
+  {
+    // Get the user objects that have been added to get the name of the PorousFlowDictator
+    auto userobjects = _awh.getActions<AddUserObjectAction>();
+    for (auto & userobject : userobjects)
+      if (userobject->getMooseObjectType() == "PorousFlowDictator")
+        _dictator_name = userobject->name();
+
+    // Get the list of materials that have been added
+    auto actions = _awh.getActions<AddMaterialAction>();
+
+    for (auto & action : actions)
+    {
+      AddMaterialAction * material = const_cast<AddMaterialAction *>(action);
+
+      // If a PorousFlowJoiner material has been included in the input file,
+      // let the user know that it should be removed
+      if (material->getMooseObjectType() == "PorousFlowJoiner")
+        mooseError("PorousFlowJoiner materials are no longer required in the input file.\nPlease "
```

Comment:
Thanks for the suggestions. I'll make this a warning instead of an error, but it will need some more logic as this action and the `PorousFlowJoiner` in an input file both add the same material property name, so will cause an error anyway.